app/middleware/cache_request.py

```diff
@@ -42,23 +42,22 @@ async def cache_response(message: Message) -> None:
 
             # If the key is not in the cache, get the response from the request and cache it.
             key = hash((rid, eid))
-            if cached := self.cache.get(key):
-                if cached["state"] == "pending":
-                    timeout_timestamp = time.time() + self.timeout
-                    # If the state is pending, wait until the state is changed to success or failed or timeout occurs.
-                    while cached := self.cache.get(key) or time.time() < timeout_timestamp:
-                        match cached["state"]:
-                            case "success":
-                                # If the state is success, return the cached response.
-                                await JSONResponse(content=cached["data"], status_code=200)(scope, receive, send)
-                                return
-                            case "failed":
-                                # If the state is failed, attempt to request again.
-                                break
-                            case "pending":
-                                # If the state is pending, wait for 0.1 seconds.
-                                await asyncio.sleep(0.1)
-                                pass
+            if self.cache.get(key):
```

Comment:
can we manage the cached by [expiring dict](https://pypi.org/project/expiringdict/#:~:text=expiringdict%20is%20a%20Python%20caching,during%20cleanup%20from%20expired%20values.)

Reply:
or [lru cache](https://stackoverflow.com/questions/31771286/python-in-memory-cache-with-time-to-live)